### DISCUSSION TITLE
Agent Status and Request inventory Unknown

### DISCUSSION THREAD

**Discussion by kintaro1981** [glpi-project/glpi-agent#403] [Q&A] [answered]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

None

### Describe the bug

I have the same issue as https://github.com/glpi-project/glpi-agent/issues/225

the status of agent and request inventory are "Unknown"

From GLPI Server I've done a `wget http://<agent-ip>:62354/status ` (edited as `<agent-ip>` was lost in the copy and paste)

and I got as output:

status: waiting

![immagine](https://github.com/glpi-project/glpi-agent/assets/2357176/a5d08d12-74e9-4cb2-8b13-f054cfdf5fd9)

fusion agent
![immagine](https://github.com/glpi-project/glpi-agent/assets/2357176/e4d6a61d-9102-46de-a0f1-4402b4b7280f)

glpi agent (migrated from fusion)
![immagine](https://github.com/glpi-project/glpi-agent/assets/2357176/737d86de-35d8-49b7-87fb-019dd56c34db)

### To reproduce

N/A

### Expected behavior

Clicking on the arrows I expect that the current status of the agent is loaded and to have the possibility force an inventory.

### Operating system

Windows

### GLPI Agent version

1.4

### GLPI version

10.0.6

### GLPIInventory plugin or FusionInventory for GLPI plugin version

Not applicable

### Additional context

GLPI inventory plugin 1.2.1

It happens the same with fusion inventory agent 2.5.2 clients.

**Comment by g-bougard** [comment]
Hi @kintaro1981 

I don't see any bug here. Have you read carefully my answer on #225 ? Have you an answer in your case for all the point I enumerated ?

You are saying you requested: `wget http://:62354/status` but here the URL is wrong as no ip or hostname is set. You need to run `wget http://10.20.4.50:62354/status` where I put the contact ip I can see in your screenshot.

You can eventually enable `debug=2` and restart the service. Then check the agent log you find a "status" request at the time you run wget.

If you have the line in the log, you should also consider to update GLPI to the latest version and try again.

**Reply by kintaro1981** [reply]
In the first message something got lost during the copy and paste:

the wget command was:
`wget http://<agent ip>:62354/status`

and specifically:
`wget http://10.20.4.50:62354/status`

The wget command was launched from the glpi server and the client replied correctly with the status of the agent `status: waiting`.
I tried also to browse http://10.20.4.50:62354/status from a browser in another client and it loaded correctly replying: `status: waiting`

**Reply by g-bougard** [reply]
Fine, so I agree your agent is doing the job. So this should be an issue from GLPI. Have you upgraded it as I suggested ?
As the problem is on GLPI side, you may check the GLPI `php-errors.loog` file for any error at the time you request GLPI status.
Indeed GLPI is looking for the agent ip or hostanme from what was discovered as network interface. Can you share the `networks` node from the output of `glpi-inventory --partial network` command ? This command should be run from an administrative console and from the glpi-agent installation folder on windows.

**Reply by kintaro1981** [reply]
We just upgraded to 10.0.7 and the problem is still there.
No error in php-errors about tha "Unknown" issue.

I think that GLPI Inventory plugin try only with hostname and don't fallback to the "Public contact address" like (like fusioninventory plugin?). I tryed with some computers I know for sure that their hostnames are defined in our internal DNS and instead of "Unknow" it respond with "Not Allowed" (so at least it seems they talk each other), for hostnames not registered in the DNS it responds with Unknown.

I can't see any option in GLPI inventory plugin to make it contact agents by ip or at least fallback to the IP.

Register all the machines in the DNS is not an option right know. :-(

This is the output of `glpi-inventory --partial network` (the virtual machine part is not the problem as this issue happens for all the machines).

```
{
   "action": "inventory",
   "content": {
      "bios": {
         "assettag": "",
         "bdate": "2022-01-19",
         "biosserial": "PC2WW7E3",
         "bmanufacturer": "LENOVO",
         "bversion": "M3AKT41A",
         "enclosureserial": "PC2WW7E3",
         "mmodel": "318E",
         "msn": "",
         "smanufacturer": "LENOVO",
         "smodel": "11JAS1AB00",
         "ssn": "PC2WW7E3"
      },
      "hardware": {
         "chassis_type": "Desktop",
         "defaultgateway": "10.2.0.200",
         "dns": "10.2.0.5",
         "memory": 15741,
         "name": "PC-004GAL3",
         "uuid": "729B205A-CCC7-11EC-90C0-332953C28600",
         "vmsystem": "Physical",
         "winlang": "1040",
         "winowner": "utente",
         "winprodid": "00XXX-X0000-00000-AAOEM",
         "winprodkey": "ZZZZZ-XXXXX-XXXXX-XXXXX-XXXXX",
         "workgroup": "SIDOM"
      },
      "networks": [
         {
            "description": "VirtualBox Host-Only Ethernet Adapter",
            "ipaddress": "192.168.56.1",
            "ipmask": "255.255.255.0",
            "ipsubnet": "192.168.56.0",
            "mac": "0A:00:27:00:00:0D",
            "pnpdeviceid": "ROOT\\NET\\0000",
            "speed": "1000",
            "status": "up",
            "virtualdev": true
         },
         {
            "description": "VirtualBox Host-Only Ethernet Adapter",
            "ipaddress6": "fe80::a7f5:1213:b1e2:96ba",
            "ipmask6": "ffff:ffff:ffff:ffff::",
            "ipsubnet6": "fe80::",
            "mac": "0A:00:27:00:00:0D",
            "pnpdeviceid": "ROOT\\NET\\0000",
            "speed": "1000",
            "status": "up",
            "virtualdev": true
         },
         {
            "description": "Realtek PCIe GbE Family Controller #2",
            "ipaddress": "10.2.15.115",
            "ipgateway": "10.2.0.200",
            "ipmask": "255.255.0.0",
            "ipsubnet": "10.2.0.0",
            "mac": "88:AE:DD:18:9B:4C",
            "pciid": "10EC:8168:318E:17AA",
            "pnpdeviceid": "PCI\\VEN_10EC&DEV_8168&SUBSYS_318E17AA&REV_0E\\6&1957c096&0&00400211",
            "speed": "100",
            "status": "up",
            "virtualdev": false
         },
         {
            "description": "Realtek PCIe GbE Family Controller #2",
            "ipaddress6": "fe80::a77f:c44a:69bf:ea25",
            "ipmask6": "ffff:ffff:ffff:ffff::",
            "ipsubnet6": "fe80::",
            "mac": "88:AE:DD:18:9B:4C",
            "pciid": "10EC:8168:318E:17AA",
            "pnpdeviceid": "PCI\\VEN_10EC&DEV_8168&SUBSYS_318E17AA&REV_0E\\6&1957c096&0&00400211",
            "speed": "100",
            "status": "up",
            "virtualdev": false
         }
      ],
      "versionclient": "GLPI-Inventory_v1.4"
   },
   "deviceid": "PC-004GAL3-2023-05-29-10-56-40",
   "itemtype": "Computer",
   "partial": true
}
```

**Reply by g-bougard** [reply]
Here I see another ip than `10.20.4.50`. Is this the same computer ? Can you reach this ip from GLPI server ? So what's the output of `wget http://10.2.15.115:62354/status` ?

Don't assume GLPI Inventory plugin is the faulty part. This is not the case as in GLPI 10 inventory is native. So the related code is really in GLPI Core, not in any plugin. GLPI Inventory plugin purpose is only to support tasks like FI plugin was.

**Reply by kintaro1981** [reply]
I simply wrote a different IP because I was testing another computer, but all the computers I tested soffer of the same problem.

`wget http://10.2.15.115:62354/status` reply with "waiting"
`wget http://10.20.4.50:62354/status` reply with "waiting"

The problem I see is that GLPI call the agent by hostname and not fallback to the IP.

`wget http://hostname:62354/status` reply with "waiting" (if the hostname is registered in the dns)

Actually Web GUI reply with:
"Unknown" for computers not registered in the DNS so in our network can not be pinged with hostname.
"Not Allowed" for computers registered in the DNS so in our network can be pinged with hostname.

I found the the meaning of the two replies here:
https://github.com/glpi-project/glpi/blob/52224e88f365c48f425264bbe09c3062abe8b1e6/src/Agent.php#L739

Now... I think that GLPI try to check the status using the hostname ony, can you confirm this?
For the "Not Allowed" message, I don't know. Is there some permission to change in GLPI configuration?

**Reply by kintaro1981** [reply, marked as answer]
SOLVED!

The problem was the configuration of the proxy inside GLPI.

I found out that `public function requestAgent` use proxy in case is configurated (and not only for upgrades as front/config.form.php say) here:
https://github.com/glpi-project/glpi/blob/52224e88f365c48f425264bbe09c3062abe8b1e6/src/Agent.php#L694

We are migrating from a "proxy" to a "no proxy" network configuration and right know both are working but with the upgrade from 9.5.x to 10.0.x I think that the behaviour of GLPI/Fusion changed.

I left a comment in the issue I linked for future "issuers".

**Reply by g-bougard** [reply]
Oh yes, thank you I remember now this also was the same problem for other people.
Thank you for that update.

**Comment by donmart85** [comment]
i have the problem, 
my version is 10.0.14 and glpi inventory GLPI-Agent_v1.7.3
if i launch from server i can see the task correctly on the agent and it works.
i can reache the web page from server, so it isn't a firewall problem.
can you help me?
Thanks.
